### PR TITLE
Enable type checking

### PR DESCRIPTION
**Pull request by xkubov**
This PR provides small but necessary changes to enable correct type checking with mypy.

### CODE REVIEW

File: enumfields/enums.py [hunk 1]
```diff
@@ -114,6 +115,10 @@ def class_to_str(cls):
 
 
 class ChoicesEnum(enum.Enum, metaclass=ChoiceEnumMeta):
+    value: Any
+    label: str
+    initial: bool
+    next: list[str] | None
```
Comment (xkubov) [inline enumfields/enums.py:121]: This is not the nicest solution but for our use case is sufficient enough.

Mypy has a problem inferring this information from the dynamic construction of the enum class. These annotations will help MyPy with the majority of our use cases. It won't work for cases where we use non-standard keys (provided through `kwargs` of `Choice`) - in these cases, we need to annotate the special keys on our side.

Comment (HonzaSaibic) [inline enumfields/enums.py:121]: next should be set(str) | None. Also label should be StrPromise

Comment (HonzaSaibic) [reply enumfields/enums.py:121]: for the label, maybe consider use of StrOrPromise. We might have some occurrences of untranslated labels.

**Review by rubickcz — APPROVED** [review]

File: enumfields/enums.py (hunk 1 above)
Comment (xkubov) [reply enumfields/enums.py:121]: I did not use StrOrPromise by choice. I think it is unnecessary to include another dependency only for this type. Mypy won't ever compare the real value passed to `Choice` constructor with this type so we can safely use `str`.

**Review by HonzaSaibic — APPROVED** [review]

**Pull request merged**